src/metatrain/utils/evaluate_model.py

```diff
@@ -320,3 +256,50 @@ def _get_model_outputs(
                 for key, value in targets.items()
             },
         )
+
+
+def _prepare_system(system: System, positions_grad: bool, strain_grad: bool):
+    """
+    Prepares a system for gradient calculation.
+    """
+    if strain_grad:
+        strain = torch.eye(
+            3,
+            requires_grad=True,
+            dtype=system.cell.dtype,
+            device=system.cell.device,
+        )
+        new_system = System(
+            positions=system.positions @ strain,
+            cell=system.cell @ strain,
+            types=system.types,
+        )
+    else:
+        if positions_grad:
+            new_system = System(
+                positions=system.positions.detach().clone().requires_grad_(True),
+                cell=system.cell,
+                types=system.types,
+            )
+            strain = None
+        else:
+            new_system = System(
+                positions=system.positions,
+                cell=system.cell,
+                types=system.types,
+            )
+            strain = None
+    for nl_options in system.known_neighbor_lists():
+        nl = system.get_neighbor_list(nl_options)
+        register_autograd_neighbors(
+            new_system,
+            TensorBlock(
+                values=nl.values.detach(),
+                samples=nl.samples,
+                components=nl.components,
+                properties=nl.properties,
+            ),
```

Comment:
why is this creating a new TensorBlock? This is what causes the next line to send the warning

Reply:
Because we have to detach. Otherwise it errors saying that the nl already requires grad

Reply:
ok, then you should do something like

```py
nl = system.get_nl()
nl = metatensor.torch.detach_block(nl)
register_autograd_neighbors(new_system, nl)
new_system.add_nl(nl)
```

This removes the warning & make sure the right graph is created without recomputing